cli/src/maintenance.rs

```diff
@@ -606,6 +623,7 @@ impl SolidoState {
 
     /// If there is a validator being deactivated, try to unstake its funds.
     pub fn try_unstake_from_inactive_validator(&self) -> Option<(Instruction, MaintenanceOutput)> {
+        self.should_perform_maintenance_at_end_of_epoch()?;
```

Comment:
This one we can leave unguarded. With unstake from active validators, it is a last resort, but with unstake from a validator that needs to be removed, we can unstake immediately.